torch/csrc/Layout.cpp

```diff
@@ -0,0 +1,88 @@
+  auto self = THPObjectPtr{type->tp_alloc(type, 0)};
+  if (!self) throw python_error();
+  auto self_ = reinterpret_cast<THPLayout*>(self.get());
+  self_->is_strided = is_strided;
+  std::strncpy (self_->name, name.c_str(), LAYOUT_NAME_LEN);
+  self_->name[LAYOUT_NAME_LEN] = '\0';
+  return self.release();
+}
+
+PyObject *THPLayout_repr(THPLayout *self)
+{
+  return THPUtils_packString(self->name);
+}
+
+PyObject *THPLayout_is_cuda(THPLayout *self)
+{
+  Py_RETURN_TRUE;
+}
+
+typedef PyObject *(*getter)(PyObject *, void *);
+
+static struct PyGetSetDef THPLayout_properties[] = {
+  {"is_cuda",      (getter)THPLayout_is_cuda, nullptr, nullptr, nullptr},
+  {nullptr}
+};
+
+PyTypeObject THPLayoutType = {
+  PyVarObject_HEAD_INIT(nullptr, 0)
+  "torch.layout",                        /* tp_name */
+  sizeof(THPLayout),                     /* tp_basicsize */
+  0,                                     /* tp_itemsize */
+  0,                                     /* tp_dealloc */
+  0,                                     /* tp_print */
+  0,                                     /* tp_getattr */
+  0,                                     /* tp_setattr */
+  0,                                     /* tp_reserved */
+  (reprfunc)THPLayout_repr,              /* tp_repr */
+  0,                                     /* tp_as_number */
+  0,                                     /* tp_as_sequence */
+  0,                                     /* tp_as_mapping */
+  0,                                     /* tp_hash  */
+  0,                                     /* tp_call */
+  0,                                     /* tp_str */
+  0,                                     /* tp_getattro */
+  0,                                     /* tp_setattro */
+  0,                                     /* tp_as_buffer */
+  Py_TPFLAGS_DEFAULT,                    /* tp_flags */
+  nullptr,                               /* tp_doc */
+  0,                                     /* tp_traverse */
+  0,                                     /* tp_clear */
+  0,                                     /* tp_richcompare */
+  0,                                     /* tp_weaklistoffset */
+  0,                                     /* tp_iter */
+  0,                                     /* tp_iternext */
+  0,                                     /* tp_methods */
+  0,                                     /* tp_members */
+  THPLayout_properties,                                     /* tp_getset */
+  0,                                     /* tp_base */
+  0,                                     /* tp_dict */
+  0,                                     /* tp_descr_get */
+  0,                                     /* tp_descr_set */
+  0,                                     /* tp_dictoffset */
+  0,                                     /* tp_init */
+  0,                                     /* tp_alloc */
+  0,                                     /* tp_new */
+};
+
+bool THPLayout_init(PyObject *module)
```

Comment:
I think it' better to have a `void` return type and throw `python_error` on exceptions. I know there are still a lot of `_init` functions where we don't do this, but when things go wrong you get more specific error messages.

Reply:
Right, I should change Dtype too otherwise I'll keep propagating the old pattern.